onnx/defs/reduction/utils.cc

```diff
@@ -61,17 +61,19 @@ to `False` instead of `True`.)DOC";
     if (axes_input) {
       schema.Attr(
           "noop_with_empty_axes",
-          "Defines behavior if 'axes' is empty. Default behavior with 'false' is to reduce all axes. "
-          "When axes is empty and this attribute is set to true, input tensor will not be reduced,"
-          "and the output tensor would be equivalent to input tensor.",
+          "Defines behavior when 'axes' is explicitly provided but empty. "
+          "If false (default), reduction happens over all axes. "
+          "If true, input tensor is returned unchanged (acts as Identity op).",
           AttributeProto::INT,
           static_cast<int64_t>(0));
       schema.Input(
           1,
           "axes",
           "Optional input list of integers, along which to reduce. "
-          "The default is to reduce over all the dimensions of the input tensor if 'noop_with_empty_axes' is false, "
-          "else act as an Identity op when 'noop_with_empty_axes' is true. "
+          "When not provided, reduction happens over all axes. "
```

Comment:
My interpretation is different. I think the default value for this input, if not explicitly specified, is the empty list. The behavior when this value is the empty list (either when explicitly specified or when omitted) is controlled by the noop_with_empty_axes attribute.